### PR TITLE
Make position determination more robust in the starlist

### PR DESCRIPTION
**Pull request by profjsb**
When the positional uncertainty is nongiven or zero, make sure that the centroid positioning for the starlist/finder returns a sensible position.

### CODE REVIEW

**Review by acrellin — APPROVED** [review]

File: skyportal/utils/offset.py [hunk 1]
```diff
@@ -285,15 +286,23 @@ def _calculate_best_position_for_offset_stars(
...
+            diff_ra = np.average(df["ra_offset"], weights=df["snr"] ** 2)
+            diff_dec = np.average(df["dec_offset"], weights=df["snr"] ** 2)
+        elif how == "invvar":
+            diff_ra = np.average(df["ra_offset"], weights=1 / df["ra_unc"] ** 2)
+            diff_dec = np.average(df["dec_offset"], weights=1 / df["dec_unc"] ** 2)
+        else:
+            log(f"Warning: do not recognize {how} as a valid way to weight astrometry")
+            return (med_ra, med_dec)
+    except ZeroDivisionError as e:
+        log(f"ZeroDivisionError in calculating position with {how}: {e}")
```
Comment (stefanv) [inline skyportal/utils/offset.py:301]: Do we want to do something special in SourceFinderHandler when these errors occur?  If so, this function may want to raise.

Comment (profjsb) [reply skyportal/utils/offset.py:301]: No, the median position will be fine enough. It's not an error they needs to be dealt with at the user level. We're already in the "likely doesn't matter" regime with the median vs the weighted average pos.

Comment (stefanv) [reply skyportal/utils/offset.py:301]: OK, great

**Comment by stefanv** [comment]
Looks like this needs a rebase, since it contains commits from the master branch.

File: skyportal/utils/offset.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 log = make_log('finder-chart')
 
 warnings.simplefilter('ignore', category=AstropyWarning)
+warnings.simplefilter('ignore', category=RuntimeWarning)
```
[the remark this answers is not in the document]
Comment (profjsb) [reply skyportal/utils/offset.py]: Done. Using a decorator to watch warnings now.

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
The patch looks good to me!  Some weird AstroPy error; related?

**Comment by profjsb** [comment]
No, I think the GA failure is something unrelated. It's happening during the requirements installation.

**Comment by stefanv** [comment]
OK, will test locally.

**Pull request merged**